tests/integration/pilot/analyze_test.go

```diff
@@ -112,33 +112,8 @@ func TestDirectoryWithoutRecursion(t *testing.T) {
 
 			istioCtl := istioctl.NewOrFail(t, istioctl.Config{})
 
-			// Recursive is false, so we should only analyze
-			// testdata/some-dir/missing-gateway.yaml and get a
-			// SchemaValidationError (if we did recurse, we'd get a
-			// UnknownAnnotation as well).
-			output, err := istioctlSafe(t, istioCtl, ns.Name(), false, "--recursive=false", dirWithConfig)
-			expectMessages(t, g, output, msg.SchemaValidationError)
-			g.Expect(err).To(BeIdenticalTo(analyzerFoundIssuesError))
-		})
-}
-
-func TestDirectoryWithRecursion(t *testing.T) {
-	// nolint: staticcheck
-	framework.
-		NewTest(t).
-		RequiresSingleCluster().
-		Run(func(t framework.TestContext) {
-			g := NewWithT(t)
-
-			ns := namespace.NewOrFail(t, namespace.Config{
-				Prefix: "istioctl-analyze",
-				Inject: true,
-			})
-
-			istioCtl := istioctl.NewOrFail(t, istioctl.Config{})
-
-			// Recursive is true, so we should see one error (SchemaValidationError).
-			output, err := istioctlSafe(t, istioCtl, ns.Name(), false, "--recursive=true", dirWithConfig)
+			// Hardcore recursive to true, so we should see one error (SchemaValidationError).
```

Comment:
`UnknownAnnotation` should be included here as well. BTW please avoid force push after reviews - Istio will do a squash merge for you once the PR is merged.

Reply:
Sorry, I don't understand what `UnknownAnnotation` means. Can you explain it to me or tell me how to modify it? Thanks.

Reply:
If you check the test directory, there's also another invalid config that should report an `UnknownAnnotation` error.

Reply:
For the test result, there don't report `UnknownAnnotation` error now, maybe we should open another PR to see how to deal with it:
https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53582/integ-pilot_istio/1849378197053902848

Reply:
This test doesn't make sense to me. Rather than keeping a test that doesn't test the feature in the codebase, I prefer to fix first.

Reply:
I guess the root cause is that the service type is default excluded in analysis, and that annotation may have been skipped by the analyzers. You may try with a different resource.

Reply:
NVM I think this actually tests something, although it's not very obvious. We can improve this later.

Reply:
I used the command line to do relevant tests yesterday and looked through the relevant code logic. I have not found any relevant scenarios where `UnknownAnnotation` is reported. I have not found any relevant reasons yet (it seems that the Annotation part will not be executed: https://github.com/istio/istio/blob/master/pkg/config/analysis/analyzers/annotations/annotations.go#L53), which requires further investigation.